modules/rostests/apitests/shlwapi/SHLoadIndirectString.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for SHLoadIndirectString
+ * COPYRIGHT:   Copyright 2020 Mark Jansen (mark.jansen@reactos.org)
+ */
+
+#include <apitest.h>
+#include <shlwapi.h>
+#include "resource.h"
+
+static void execute_test(LPCWSTR DllFile)
+{
+    WCHAR DllBuffer[MAX_PATH + 20];
+    WCHAR Buffer[MAX_PATH * 2] = {0};
+    HRESULT hr;
+    HANDLE hEvent;
+    DWORD dwRet;
+
+    hEvent = CreateEventA(NULL, TRUE, FALSE, "Local\\shlwapi_apitest_evt");
+
+    // Show that it is not signaled
+    dwRet = WaitForSingleObject(hEvent, 1);
+    ok_hex(dwRet, WAIT_TIMEOUT);
+
+    // Ensure the module is not loaded yet...
+    HMODULE mod = GetModuleHandleW(DllFile);
+    if (mod != NULL)
+    {
+        skip("%S loaded, cannot continue\n", DllFile);
+        return;
+    }
+
+    wsprintfW(DllBuffer, L"@%s,-3", DllFile);
+
+    // Load a string from the dll
+    hr = SHLoadIndirectString(DllBuffer, Buffer, _countof(Buffer), NULL);
+    ok_hex(hr, S_OK);
+    if (SUCCEEDED(hr))
+    {
+        // Module should still not be loaded
+        mod = GetModuleHandleW(DllFile);
+        ok_ptr(mod, 0);
+
+        ok_wstr(Buffer, L"Test string one.");
+
+        // DllMain not called..
+        dwRet = WaitForSingleObject(hEvent, 1);
+        ok_hex(dwRet, WAIT_TIMEOUT);
+
+        // Show that calling DllMain will set the event
+        mod = LoadLibraryW(DllFile);
+        ok(mod != NULL, "Failed to load %S\n", DllFile);
+
+        if (mod)
+        {
+            dwRet = WaitForSingleObject(hEvent, 1);
+            ok_hex(dwRet, WAIT_OBJECT_0);
+            FreeLibrary(mod);
+        }
+    }
+    CloseHandle(hEvent);
+}
+
+
+BOOL extract_resource(const WCHAR* Filename, LPCWSTR ResourceName)
```

Comment:
Make the function static

Reply:
No, other apitests can share the function if it's not static.